### ISSUE TITLE
No type error in function called as argument of another function which accepts `Any`

### ISSUE THREAD

**Issue by layday** [microsoft/pyright#2313]
**Describe the bug**
For a callable which takes `Any`, calling another callable in the `Any` parameter's position will not produce a type error where it otherwise would, if both the argument and the return value of the other callable are containers parametrised with a bound type variable.

**To Reproduce**
Type check the snippet below.

**Expected behavior**
That Pyright would produce an error on line 13.

**Screenshots or Code**
```python
# pyright: strict

from typing import Any, Iterable, TypeVar

String = TypeVar("String", bound=str)

def foo(value: Any):
    ...

def bar(value: Iterable[String]) -> list[String]:
    ...

foo(bar([1, 2, 3]))  # Should error but it doesn't
bar([1, 2, 3])  # Errors
bar(["1", "2", "3"])  # Ok
```

**VS Code extension or command-line**
v1.1.169

**Comment by erictraut** [comment]
Thanks for the bug report. This will be fixed in the next release.

**Comment by erictraut** [comment]
This is addressed in pyright 1.1.170, which I just published. It will also be included in the next release of pylance.
